src/btree/bt_read.c

```diff
@@ -114,12 +114,25 @@ __las_page_instantiate(WT_SESSION_IMPL *session, WT_REF *ref, uint32_t btree_id)
 	 * updates for the next key, and so on.
 	 *
 	 * Search for the block's unique prefix, stepping through any matching
-	 * records.
+	 * records.  Because of the special visibility rules for lookaside, a
+	 * new block can appear in between our search and the block of
+	 * interest.  Keep trying until we find it.
```

Comment:
I'd suggest updating both versions of this comment to note the lookaside block may not appear at all, and why.

If you're willing, I'd even abstract this code out to a `__wt_las_cursor_position` call, given that it's relatively obscure what's happening.

Reply:
Thanks, agreed -- done.